extensions/interactions/base.py

```diff
@@ -89,6 +89,11 @@ class BaseInteraction(object):
     # containing this interaction. These should correspond to names of files in
     # feconf.DEPENDENCIES_TEMPLATES_DIR. Overridden in subclasses.
     _dependency_ids = []
+    # Additional JS library dependencies that should be loaded in pages
```

Comment:
So the issue is with prediction service registry, right? We can't inject the deps in factory hardcoded way, cause that will raise exceptions if some of them are missing. So we have to do dynamic injection there as well. I'll have to research a bit whether it is possible to do or not and how.